packages/utils/src/metrics.ts

```diff
@@ -0,0 +1,80 @@
+export type LabelsGeneric = Record<string, string | number>;
+export type LabelKeys<Labels extends LabelsGeneric> = Extract<keyof Labels, string>;
```

Comment:
Default prom-client label type is `<T extends string = string>` if you just do `keyof Labels` here then it will complain

```
Type 'keyof Labels' does not satisfy the constraint 'string'.
  Type 'string | number | symbol' is not assignable to type 'string'.
    Type 'number' is not assignable to type 'string'.ts(2344)
```

I have not found another way to resolve this other than using `Extract`